Mondee Watch App/ViewModel/DeviceCommunicator.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  DeviceCommunicator.swift
+//  Mondee
+//
+//  Created by hyunjun on 2023/07/15.
+//
+
+import Foundation
+import WatchConnectivity
+
+/// Two-way communication between an iOS app and its paired watchOS app.
+/// 🚨 Message enum is required
+class DeviceCommunicator: NSObject, WCSessionDelegate {
+    private var session: WCSession = .default
+    
+    /// Data Recieved handler
+    var dataReceived: ((MessageKey, Any) -> Void)?
+    
+    init(session: WCSession = .default) {
+        self.session = session
+        
+        super.init()
+        
+        self.session.delegate = self
+        self.connect()
+    }
+    
+    /// Activates the session if WCSession is supported.
+    private func connect() {
+        guard WCSession.isSupported() else {
+            print("WCSession is not supported")
+            return
+        }
+        session.activate()
+    }
+    
+    func session(_ session: WCSession, activationDidCompleteWith activationState: WCSessionActivationState, error: Error?) { }
+    
+#if os(iOS)
+    func sessionDidBecomeInactive(_ session: WCSession) { }
+    
+    func sessionDidDeactivate(_ session: WCSession) { }
+#endif
+    
+    /// Handles the received message.
+    func session(_ session: WCSession, didReceiveMessage message: [String: Any]) {
```

Comment:
노션에 정리해주신 글을 봤는데 애플에서 정해준 함수이기때문에 함수이름을 변경을 못할까요?🥹
함수 이름이 더 직관적이면 좋을 것 같아요! ex) receiveMessage 등
Suggestion 한 번 해보겠습니다!

```suggestion
    func recevieMessage(_ session: WCSession, didReceiveMessage message: [String: Any]) {
```

Reply:
헉 불망 멋지시군요. 저는 생각도 안해봤던것 같아요. 찾아보니 `WCSessionDelegate` 프로토콜을 따르는 이상 어쩔수 없는 것 같습니다. 그리고 위에 `self.session.delegate = self`, `session.activate()` 같은 메서드와 연결되어 있어서 변경은 어려울 것 같아요!!

Reply:
아하 프로토콜을 따르는 군요! 그러면 어쩔수없네요ㅠ 프로토콜을 보니 session함수안에 didReceiveMessage 변수를 사용하는군요!
설명 감사합니다~!